src/test/custom_nodes/node_choose_maximum.cpp

```diff
@@ -0,0 +1,172 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+
+extern "C" {
+enum Method {
+    MAXIMUM_MINIMUM,
+    MAXIMUM_AVERAGE,
+    MAXIMUM_MAXIMUM,
+    METHOD_COUNT
+};
+
+static const std::string INPUT_TENSOR_NAME = "input_tensors";
+
+int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength) {
+    // choose selection criteria
+    Method selectionMethod = Method::METHOD_COUNT;
+    if (paramsLength != 1) {
+        std::cout << "Wrong number of parameters - expected 1" << std::endl;
+        return 1;
+    }
+    if (strcmp(params[0].key, "selection_criteria") == 0) {
+        if (strcmp(params[0].value, "MAXIMUM_MINIMUM") == Method::MAXIMUM_MINIMUM) {
+            selectionMethod = Method::MAXIMUM_MINIMUM;
+        } else if (strcmp(params[0].value, "MAXIMUM_MAXIMUM") == Method::MAXIMUM_MAXIMUM) {
+            selectionMethod = Method::MAXIMUM_MAXIMUM;
+        } else if (strcmp(params[0].value, "MAXIMUM_AVERAGE") == Method::MAXIMUM_AVERAGE) {
+            selectionMethod = Method::MAXIMUM_AVERAGE;
+        } else {
+            std::cout << "Not allowed selection criteria chosen:" << params[0].value << std::endl;
+            return 1;
+        }
+    } else {
+        std::cout << "Non recognized param string" << std::endl;
+        return 1;
+    }
+    // get input tensor
+    if (inputsLength != 1) {
+        std::cout << "Wrong number of inputs - expected 1" << std::endl;
+        return 1;
+    }
+    float* inputTensor;
+    size_t valuesPerTensor = 0;
+    size_t numberOfOps = 0;
+    for (size_t i = 0; i < inputsLength; ++i) {
+        if (INPUT_TENSOR_NAME == inputs[i].name) {
+            if (inputs[i].dimsLength != 3 ||
+                inputs[i].dims[0] != 1 ||
+                inputs[i].dims[1] == 0 ||
+                inputs[i].dims[2] == 0) {
+                std::cout << "improper " << INPUT_TENSOR_NAME
+                          << " dimensions: [" << inputs[i].dims[0]
+                          << ", " << inputs[i].dims[1]
+                          << ", " << inputs[i].dims[2] << "]" << std::endl;
+                return 1;
+            }
+            std::cout << "Input valuesPerTensor" << inputs[i].dims[1] << std::endl;
+            numberOfOps = inputs[i].dims[1];
+            valuesPerTensor = inputs[i].dims[2];
+            inputTensor = reinterpret_cast<float*>(inputs[i].data);
+        } else {
+            std::cout << "Lacking input:" << INPUT_TENSOR_NAME << std::endl;
+            return 1;  // TODO free
```

Comment:
Not neccessary. We do not need to free inputs, ovms always takes care of that.

Reply:
Yes - missed that TODO - removed.